src/rules.js

```diff
@@ -74,7 +74,7 @@ const rules = {
    */
   validate: (def, data, key = null, errors = [], rejectOnFail = true) => {
     let curData = data
-    const props = !def.required && data === undefined ? rules.props.noVal : rules.props.default
+    const props = !def.required && data === undefined || def.allow === null ? rules.props.noVal : rules.props.default
```

Comment:
This should also check that the value (`data`) is `null`, if it's _not_ we want it to run type validations on it.